pype/plugins/blender/load/load_rig.py

```diff
@@ -0,0 +1,256 @@
+"""Load a rig asset in Blender."""
+
+import logging
+from pathlib import Path
+from pprint import pformat
+from typing import Dict, List, Optional
+
+from avalon import api, blender
+import bpy
+import pype.blender.plugin
+
+logger = logging.getLogger("pype").getChild("blender").getChild("load_model")
+
+
+class BlendRigLoader(pype.blender.plugin.AssetLoader):
+    """Load rigs from a .blend file.
+
+    Because they come from a .blend file we can simply link the collection that
+    contains the model. There is no further need to 'containerise' it.
+
+    Warning:
+        Loading the same asset more then once is not properly supported at the
+        moment.
+    """
+
+    families = ["rig"]
+    representations = ["blend"]
+
+    label = "Link Rig"
+    icon = "code-fork"
+    color = "orange"
+
+    @staticmethod
+    def _remove(self, objects, lib_container):
+
+        for obj in objects:
+
+            if obj.type == 'ARMATURE':
+                bpy.data.armatures.remove(obj.data)
+            elif obj.type == 'MESH':
+                bpy.data.meshes.remove(obj.data)
+
+        bpy.data.collections.remove(bpy.data.collections[lib_container])
+
+    @staticmethod
+    def _process(self, libpath, lib_container, container_name, action):
+
+        relative = bpy.context.preferences.filepaths.use_relative_paths
+        with bpy.data.libraries.load(
+            libpath, link=True, relative=relative
+        ) as (_, data_to):
+            data_to.collections = [lib_container]
+
+        scene = bpy.context.scene
+
+        scene.collection.children.link(bpy.data.collections[lib_container])
+
+        rig_container = scene.collection.children[lib_container].make_local()
+
+        meshes = [obj for obj in rig_container.objects if obj.type == 'MESH']
+        armatures = [
+            obj for obj in rig_container.objects if obj.type == 'ARMATURE']
+
+        objects_list = []
+
+        assert(len(armatures) == 1)
```

Comment:
why this assert? It should throw some error message and there should be corresponding validator if this is enforced

Reply:
we'll make a ticket for this. but merging for now, as this is good enough for testing, to make on it's way to production.